basic-examples/bind-constants/workflow.yaml

```diff
@@ -0,0 +1,43 @@
+version: v0.2.0
+input:
+  root: RootObject
+  objects:
+    RootObject:
+      id: RootObject
+      properties:
+        repeated_inputs:
+          type:
+            type_id: ref
+            id: RepeatedValues
+        iterations:
+          type:
+            type_id: list
+            items:
+              id: SubRootObject
+              type_id: ref
+    RepeatedValues:
+      id: RepeatedValues
+      properties:
+        name:
+          type:
+            type_id: string
+        ratio:
+          type:
+            type_id: float
+    SubRootObject:
+      id: SubRootObject
+      properties:
+        loop_id:
+          type:
+            type_id: integer            
+            
+steps:
+  foreach_loop:
+    kind: foreach
+    items: !expr 'bindConstants($.input.iterations, $.input.repeated_inputs)'
+    workflow: subworkflow.yaml
+    parallelism: 1
+
+outputs:
+  success:
+    fab_four: !expr $.steps.foreach_loop.outputs.success.data

```

Comment:
It might be worth having a comment here explaining what we expect to see returned from the subworkflow:  a list containing entries for the output of each iteration which, in this case, contains the same information that we sent in -- an object containing the `loop_id` and the `repeated_values` subobject.

Reply:
this is what the output looks like

```yaml
output_data:
    fab_four:
        - beatle:
            message: Hello, mogo!
          loop_id: 1
          ratio: 3.14
        - beatle:
            message: Hello, mogo!
          loop_id: 2
          ratio: 0.5
        - beatle:
            message: Hello, Matt!
          loop_id: 3
          ratio: 3.14
        - beatle:
            message: Hello, mogo!
          loop_id: 4
          ratio: 3.14
output_id: success
```